jina/serve/runtimes/request_handlers/data_request_handler.py

```diff
@@ -10,28 +10,40 @@
 if TYPE_CHECKING:
     import argparse
 
+    from prometheus_client import CollectorRegistry
+
     from jina.logging.logger import JinaLogger
 
 
 class DataRequestHandler:
     """Object to encapsulate the code related to handle the data requests passing to executor and its returned values"""
 
-    def __init__(self, args: 'argparse.Namespace', logger: 'JinaLogger', **kwargs):
+    def __init__(
+        self,
+        args: 'argparse.Namespace',
+        logger: 'JinaLogger',
+        metrics_registry: Optional['CollectorRegistry'] = None,
+        **kwargs,
+    ):
         """Initialize private parameters and execute private loading functions.
 
         :param args: args from CLI
         :param logger: the logger provided by the user
+        :param metrics_registry: optional metrics registry for prometheus used if we need to expose metrics from the executor of from the data request handler
         :param kwargs: extra keyword arguments
         """
         super().__init__()
         self.args = args
         self.args.parallel = self.args.shards
         self.logger = logger
         self._is_closed = False
-        self._load_executor()
+        self._load_executor(metrics_registry)
```

Comment:
Since when `Executor` requires or has a `metrics_registry` attribute? It is wrong for an external entity to add attributes to an object.